### PR TITLE
core: clean up methods and tests

### PR DESCRIPTION
**Pull request by ryanchristo**
## Changes

- migrate `CompareAndSetBlockData` method to block state
- organize core service methods and tests within core service

## Tests:
```
go test ./... -short
```

### CODE REVIEW

File: dot/core/runtime.go [hunk 1]
```diff
@@ -57,7 +67,39 @@ func (s *Service) executeBlock(b []byte) error {
 	return nil
 }
 
-// TODO: this seems to be out-of-date, the call is now named Grandpa_authorities and takes a block number.
+// checkForRuntimeChanges checks if changes to the runtime code have occurred; if so, load the new runtime
```
Comment (noot) [inline dot/core/runtime.go]: would be nice to group all the runtime call functions together

File: dot/core/state_test.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package core
```
Comment (noot) [inline dot/core/state_test.go]: remove this

File: dot/state/block.go [hunk 3]
```diff
@@ -322,6 +322,46 @@ func (bs *BlockState) SetBlockData(blockData *types.BlockData) error {
 	return err
 }
 
+// CompareAndSetBlockData patches and updates the block data stored in the state
+// database with the block data provided
+func (bs *BlockState) CompareAndSetBlockData(bd *types.BlockData) error {
+	if bs == nil {
```
Comment (ryanchristo) [inline dot/state/block.go]: remove this

File: dot/core/state_test.go (hunk 2 above)
Comment (ryanchristo) [reply dot/core/state_test.go]: will do

**Comment by ryanchristo** [comment]
will continue after #708 is merged

**Pull request closed without merging**